frontend/tests/unit/stores/settings.spec.ts

```diff
@@ -0,0 +1,30 @@
+import '~/utils/mocks/globals';
+
+import { createPinia, setActivePinia } from 'pinia';
+
+import { useSettings } from '@/stores/settings';
+import { WalletType } from '@/types/settings';
+
+describe('settings store', () => {
+  beforeEach(() => {
+    const pinia = createPinia();
+    setActivePinia(pinia);
+  });
+
+  describe('setConnectedWallet', () => {
+    it('can set a connected wallet', () => {
+      const settings = useSettings();
+      settings.setConnectedWallet(WalletType.Metamask);
+      expect(settings.$state.settings?.connectedWallet).toBe(WalletType.Metamask);
+    });
```

Comment:
I expect the action to become removed. So in this case this comment becomes useless. But anyway: you should check first if `connectedWallet` is undefined. Else the execution of `setConnectedWallet` is not correctly tested as this test does not implicitly set or verify that it was a different value before calling the function.

So either add a `expect(settings.settings?.connectedWallet).toBeUndefined()` or you do `settings.settings = { connectedWallet = undefined }`. The second has the advantage to make the test work even we should decide to (for what ever reason) have a different default state.

Reply:
It was already there and has been removed an idiot move! will add it again! Thank you!